vss-accessor/tests/tests.rs

```diff
@@ -0,0 +1,272 @@
+#[cfg(test)]
+mod tests {
+	use mockito::{self, Matcher};
+	use prost::Message;
+	use vss_accessor::vss::{
+		ErrorCode, ErrorResponse, GetObjectRequest, GetObjectResponse, KeyValue, ListKeyVersionsRequest,
+		ListKeyVersionsResponse, PutObjectRequest, PutObjectResponse,
+	};
+	use vss_accessor::vss_error::VssError;
+	use vss_accessor::VssAccessor;
+
+	#[tokio::test]
+	async fn test_get() {
+		// Spin-up mock server with mock response for given request.
+		let base_url = mockito::server_url().to_string();
+		let mock_get_endpoint = "/getObject";
+
+		// Set up the mock request/response.
+		let mock_request = GetObjectRequest { store_id: "store".to_string(), key: "k1".to_string() };
+		let mut mock_response = GetObjectResponse::default();
+		mock_response.value = Some(KeyValue { key: "k1".to_string(), version: 2, value: b"k1v2".to_vec() });
+
+		// Register the mock endpoint with the mockito server.
+		let mock_server = mockito::mock("POST", mock_get_endpoint)
+			.match_body(mock_request.encode_to_vec())
+			.with_status(200)
+			.with_body(mock_response.encode_to_vec())
+			.create();
+
+		// Create a new VssAccessor with the mock server URL.
+		let vss_acc = VssAccessor::new(&base_url).unwrap();
+		let actual_result = vss_acc.get("store", "k1").await.unwrap();
+
+		let expected_result = &mock_response;
+		assert_eq!(&actual_result, expected_result);
+
+		// Verify server endpoint was called exactly once.
+		mock_server.expect(1).assert();
+	}
+
+	#[tokio::test]
+	async fn test_put() {
+		// Spin-up mock server with mock response for given request.
+		let base_url = mockito::server_url().to_string();
+		let mock_put_endpoint = "/putObjects";
+
+		// Set up the mock request/response.
+		let mock_request = PutObjectRequest {
+			store_id: "store".to_string(),
+			global_version: Some(4),
+			transaction_items: vec![KeyValue { key: "k1".to_string(), version: 2, value: b"k1v3".to_vec() }],
+		};
+		let mock_response = PutObjectResponse::default();
+
+		// Register the mock endpoint with the mockito server.
+		let mock_server = mockito::mock("POST", mock_put_endpoint)
+			.match_body(mock_request.encode_to_vec())
+			.with_status(200)
+			.with_body(mock_response.encode_to_vec())
+			.create();
+
+		// Create a new VssAccessor with the mock server URL.
+		let vss_acc = VssAccessor::new(&base_url).unwrap();
+		let actual_result = vss_acc.put("store", Some(4), "k1", 2, b"k1v3").await.unwrap();
+
+		let expected_result = &mock_response;
+		assert_eq!(&actual_result, expected_result);
+
+		// Verify server endpoint was called exactly once.
+		mock_server.expect(1).assert();
+	}
+
+	#[tokio::test]
+	async fn test_put_tx() {
+		// Spin-up mock server with mock response for given request.
+		let base_url = mockito::server_url().to_string();
+		let mock_put_endpoint = "/putObjects";
```

Comment:
Currently tests are in separate module and test using public methods, so crate wide consts doesn't work. 
I want to keep these tests segregated.

Reply:
Ok, extracted them to test-wide consts.
My intention for having them as plain strings was mainly because they were being used only once, but i don't mind either way.